omero/developers/scripts/style-guide.rst

```diff
@@ -131,6 +131,20 @@ Script outputs
 
            client.setOutput("File_Annotation",robject(fileAnnotation))
 
+-  Scripts that generate a URL link should return the omero.rtypes.rmap,
+   with the following keys: "type": "URL", "href": "URL address to open",
+   "title": "Help message". The client will give users the option
+   of opening the URL in a new browser window/tab. To use this feature 
+   the URL omero.types.rmap should use the key: 'URL' in the output map.
+
+   ::
+
+           url = omero.rtypes.rmap()
+           url.put("type", omero.rtypes.rstring("URL"))
```

Comment:
I'm not sure why, but it seems you need `url.put("id", rstring("-1"))` as in your example script.